core/test-api/src/main/kotlin/testApi/testRunner/TestDokkaConfigurationBuilder.kt

```diff
@@ -0,0 +1,151 @@
+package testApi.testRunner
+
+import org.jetbrains.dokka.*
+import org.jetbrains.dokka.links.DRI
+import org.jetbrains.dokka.model.*
+import org.jetbrains.dokka.model.doc.Description
+import org.jetbrains.dokka.model.doc.DocumentationNode
+import org.jetbrains.dokka.model.doc.Text
+import java.io.File
+
+fun dokkaConfiguration(block: TestDokkaConfigurationBuilder.() -> Unit): DokkaConfigurationImpl =
+    TestDokkaConfigurationBuilder().apply(block).build()
+
+@DslMarker
+annotation class DokkaConfigurationDsl
+
+@DokkaConfigurationDsl
+class TestDokkaConfigurationBuilder {
+
+    var moduleName: String = "root"
+        set(value) {
+            check(lazySourceSets.isEmpty()) { "Cannot set moduleName after adding source sets" }
+            field = value
+        }
+
+    var outputDir: String = "out"
+    var format: String = "html"
+    var offlineMode: Boolean = false
+    var cacheRoot: String? = null
+    var pluginsClasspath: List<File> = emptyList()
+    var pluginsConfigurations: Map<String, String> = emptyMap()
+    var failOnWarning: Boolean = false
+    private val lazySourceSets = mutableListOf<Lazy<DokkaSourceSetImpl>>()
```

Comment:
I think few things can be simplified if we were storing here not `Lazy<DokkaSourceSetImpl>` but `(moduleName: String) -> DokkaSourceSetImpl`. Especially we would get rid of strange restriction that module name must be defined before the sourcesets.

Reply:
Bump

Reply:
Oh. We should have continued this conversation here. Pawel and me talked about this in Slack: 
Basically my argument here was, that making it the signature () -> DokkaSouceSetImpl would work, but it would just obfuscate the original contract which is: Yes this is lazy, but yes you will always get the same instance. IIRC, then we agreed on using Lazy here to signal this contract!